core/include/thread.h

```diff
@@ -22,18 +22,76 @@
 #ifndef THREAD_H
 #define THREAD_H
 
-#include "kernel.h"
-#include "tcb.h"
+#include "priority_queue.h"
+#include "clist.h"
+#include "cib.h"
+#include "msg.h"
 #include "arch/thread_arch.h"
+#include "cpu_conf.h"
 
 #ifdef __cplusplus
  extern "C" {
 #endif
 
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+/**
+ * @brief Thread status list
+ * @{
+ */
+#define STATUS_NOT_FOUND        (-1)            /**< Describes an illegal thread status */
```

Comment:
Sorry for just writing that now, but wouldn't a `THREAD_` prefix on those be more in line with the coding conventions?